### PR TITLE
docs: update doc for 3.4 release

### PR DESCRIPTION
**Pull request by anupriya0703**
Signed-off-by: anupriya0703 <anupriya.gupta@mayadata.io>

### CODE REVIEW

File: docs/versioned_docs/version-3.4.x/introduction/releases.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  - [cStor (non-csi)](https://github.com/openebs/maya) 2.12.2
+
+OpenEBS Release notes are maintained in the GitHub repositories alongside the code and releases. For summary of what changes across all components in each release, checkout: https://github.com/openebs/openebs/releases
+
+Here is a quick reference on status of OpenEBS volumes and compatibility with regards to Kubernetes version. It is possible that the individual storage engines might have more dependenices. Please refer to the release notes or reach out to the [contributor community via slack](/docs/introduction/community).
+
+
+|OpenEBS Release |Kubernetes Versions|Stable Engines| Beta Engines | Alpha Engines | Deprecated Engines
+|:------ |:------------------|:--------     |:------       |:-----         |:-------
+|3.4.0 |1.18< K8s <1.25 (Jiva-CSI: K8s >=1.21) |cStor(w/CSI), LocalPV(Hostpath, Device, ZFS, LVM)| Jiva(w/CSI), Mayastor, NFS, LocalPV(Rawfile) | LocalPV(Device w/CSI) | **cStor and Jiva (w/non-CSI)**
```
Comment (niladrih) [inline docs/versioned_docs/version-3.4.x/introduction/releases.md:42]:
```suggestion
|3.4.0 |K8s >=1.21 |cStor(w/CSI), LocalPV(Hostpath, Device, ZFS, LVM)| Jiva(w/CSI), Mayastor, NFS, LocalPV(Rawfile) | LocalPV(Device w/CSI) | **cStor and Jiva (w/non-CSI)**
```

**Review by niladrih — COMMENTED** [review]
3.4.0 cStor requires K8s APIs from 1.21.
LGTM otherwise!

File: docs/versioned_docs/version-3.4.x/introduction/releases.md (hunk 1 above)
Comment (anupriya0703) [reply docs/versioned_docs/version-3.4.x/introduction/releases.md]: Updated.

**Review by niladrih — APPROVED** [review]

**Review by Pallavi-PH — APPROVED** [review]

**Review by avishnu — APPROVED** [review]
lgtm

**Pull request merged**